### DISCUSSION TITLE
Model source code

### DISCUSSION THREAD

**Discussion by csukuangfj** [snakers4/silero-vad#371] [Q&A]
There are only torchscript and onnx models, where can I find the source code of the model?

**Comment by snakers4** [comment]
Not yet

**Reply by csukuangfj** [reply]
Is there a plan to open-source the code?

**Reply by csukuangfj** [reply]
And also, do you plan to make the training code public and describe how the model is trained and how the training data is prepared?

**Reply by snakers4** [reply]
In `v5` tuning code will be made public.

**Reply by csukuangfj** [reply]
v4 was released last October. Is v5 coming soon?

**Reply by csukuangfj** [reply]
v5 is finally released. Congratulations!

Could you tell us where we can find the model source code?

**Reply by dayfinihzy** [reply]
> In `v5` tuning code will be made public.

Could we find the tuning code now？ Looking forward to receiving your answer. Thank you!

**Reply by csukuangfj** [reply]
> In `v5` tuning code will be made public.

Hi, 9 months have passed since v5 is released. Is the model source code still planned to be open-sourced?

**Comment by csukuangfj** [comment]
I have reverse engineered the PyTorch source code for silero-vad v4. Please see
https://github.com/snakers4/silero-vad/issues/630#issuecomment-3062488943
